readme.md
# ВНИМАНИЕ!!!
 Данный репозитарий является **КОПИЕЙ** оригинального репозитария [TorrentMonitor](https://github.com/ElizarovEugene/TorrentMonitor), который был созан [Евгением Елизаровым](https://github.com/ElizarovEugene).

В связи с закрытие оригинального репозитария на GitHub, было принято решение о создании репозитария со всеми доработками,  которые не попали в оригинал.

По всем вопросам просьба не беспокоить обращаться в раздел *Issues* на текущего репозитария

# TorrentMonitor
Приложение мониторит изменения на популярных торрент-трекерах рунета и автоматизирует закачку обновлений (сериалы, раздачи которые ведутся *путем добавления новых серий/новых версий*, перезалитые торрент-файлы и т.д.)

###Список возможностей приложения:

* Слежение за темами на anidub.com
* Слежение за темами на animelayer.ru
* Слежение за темами на casstudio.tv
* Слежение за темами на kinozal.tv
* Слежение за темами на nnm-club.ru
* Слежение за темами на pornolab.net
* Слежение за темами на rustorka.com
* Слежение за темами на rutracker.org
* Слежение за темами на rutor.org
* Слежение за темами на tfile.me
* Слежение за темами на tracker.0day.kiev.ua
* Слежение за релизерами на nnm-club.ru
* Слежение за релизерами на pornolab.net
* Слежение за релизерами на rutracker.org
* Слежение за релизерами на tfile.me
* Поиск новых серий на baibako.tv (SD/HD 720/HD 1080 версии на выбор)
* Поиск новых серий на lostfilm.tv (SD/HD 720/HD 1080 версии на выбор)
* Поиск новых серий на newstudio.tv (SD/HD 720/HD 1080 версии на выбор)
* Поиск новых серий на novafilm.tv (SD/HD 720 версии на выбор)

* Работа через proxy (к примеру Tor)

* Управление торрент-клиентами (добавление/удаление раздач и файлов): Transmission (через XML-RPC), Deluge (через deluge-console).

* Уведомления на E-mail и через сервисы push-уведомлений

* RSS-лента

###Требования для установки:

* Веб-сервер (Apache, nginx, lighttpd)
* PHP (5.2 или выше) с поддержкой cURL и PDO
* MySQL, PostgreSQL, SQLite

###Установка:

* Импортировать дамп базы из директории db_schema в зависимости от используемой БД - *.sql
* Перенести все файлы в папку на вашем сервере (например /path/to/folder/torrent_monitor/)
* Внести изменения в config.php и указать данные для доступа к БД

Для MySQL:
```
Config::write('db.host', 'localhost');
Config::write('db.type', 'mysql');
Config::write('db.charset', 'utf8');
Config::write('db.port', '3306');
Config::write('db.basename', 'torrentmonitor');
Config::write('db.user', 'torrentmonitor');
Config::write('db.password', 'torrentmonitor');
```
Для PostgreSQL:
```
Config::write('db.host', 'localhost');
Config::write('db.type', 'pgsql');
Config::write('db.port', '5432');
Config::write('db.basename', 'torrentmonitor');
Config::write('db.user', 'torrentmonitor');
Config::write('db.password', 'torrentmonitor');
```
Для SQLite:
```
Config::write('db.type', 'sqlite');
Config::write('db.basename', '/var/www/htdocs/TorrentMonitor/torrentmonitor.sqlite'); #Указывайте _абсолютный_ путь до файла с базой и не забудьте выставить на него верные права доступа.
```

* Добавить в cron engine.php ( *проверьте права на запись в каталог /path/to/log/* )

```
*/10 * * * * php -q /path/to/folder/torrent_monitor/engine.php >> /path/to/log/torrent_monitor_error.log 2>&1
```
* Зайти в веб-интерфейс ( **пароль по умолчанию — torrentmonitor, смените(!) его после первого входа** )
* Указать учётные данные от трекеров
* Указать в настройках путь для сохранения торрентов (папка, которая мониторится вашим торрент-клиентом), e-mail и разрешить/запретить отправку 
уведомлений
* Добавить торренты для мониторинга
* Перейти на вкладку «Тест» и проверить — всё ли верно работает

###Настройки:

Так же, в php.ini (для CLI) необходимо изменить следующие параметры:

```
; увеличить максимальное вермя выполнения скрипта
max_execution_time = 300

; указать date.timezone
date.timezone = Europe/Moscow

; эту опцию желательно включить в php.ini как для CLI, так и для веб-сервера
allow_url_fopen = on

; проверить - разрешена ли запись в сторонние каталоги. 
; Нужно разрешить запись в каталог с самим приложением TorrentMonitor 
; и каталог куда будут сохраняться *.torrent файлы для torrent клиента
open_basedir = /tmp/:/path/to/folder/torrent_monitor/:/path/to/folder/torrent_client_watch/
```


changelog.txt
15.04.2015 - 1.2.2:
- Зеркало для трекера LostFilm.TV.
- Исправлена авторизация на трекере kinozal.tv.
- Исправлены мелкие баги в интерфейсе.
- Вывод уведомлений теперь производится при помощи ohsnap https://github.com/justindomingue/ohSnap
30.03.2015 - 1.2.1:
- Корректирующий релиз для корректной работы LostFilm.TV.
23.03.2015 - 1.2:
- Вернулся LostFilm.TV благодаря пользователям GitHub, в том числе авторизация и получение всех необходимых кук.
- Улучшенная обработка ошибок для Transmission и формах
- Изменен алгоритм формирования RSS ленты 
- Фикс проблемы безопасности 
- Множество мелких исправления
12.03.2015 - 1.1.1:
- Исправлена проблема с закачкой torrent-файлов на kinozal.tv
03.03.2015 - 1.1:
- LostFilm.TV удалён из TM из-за постоянных проблем с авторизацией и чтением RSS ленты из-за cloudflare.
- Исправлены мелкие ошибки
- Использование TransmissionRPC.class
- Подключен сервис https://pushover.net
- Изменения в интерфейсе и его логике работы
- Добавлен Help
- Добавлена возможность отправки важных новостей от разработчика в ТМ
- Добавлена RSS-лента
24.10.2014 - 1.0.4:
- Исправлены мелкие ошибки
21.10.2014 - 1.0.3:
- Восстановлена работа rutor.org
18.10.2014 - 1.0.2:
- Авторизация на lostfiml.tv восстановлена
- Множество мелких исправлений
- Багфикс сохранения torrent-файлов при недоступности torrent-клиента
7.07.2014 - 1.0.1:
- Вернулись на родную RSS novafilm.tv
- Подтверждение удаления мониторинга
30.05.2014 — 1.0.0:
- Добавлен трекер pornolab.net
- Добавлен автоматический апдейт системы
27.05.2014 — 0.9.9:
- Обновление для Anidub.com: теперь в теме содержится несколько вариантов качества, система сама выбирает наилучший из имеющихся (1080 или 720, в зависимости от того, что есть, предпочтение отдаётся 1080)
- Добавлена верная иконка для страницы ручного запуска
25.05.2014 — 0.9.8:
- Обновление дизайна на Anidub.com
- Мелкие исправления
1.05.2014 — 0.9.7:
- Для Novafilm добавлена возможность скачивания сериалов в 1080
- Устранены мелкие, критические ошибки
30.04.2014 — 0.9.6:
- Добавлен трекер rustorka.com
- Добавлен трекер tracker.0day.kiev.ua
- Добавлен «буфер» для torrent-файлов, не попавших по какой то причине в torrent-клиент (если он, к примеру, был выключен) и буду автоматически добавлены как только он снова заработает
- Временное исправление для LostFilm.TV. В базе у вас должна быть кука следующего вида uid=*;pass=*;usess=*;
http://monosnap.com/image/eV5evfqXUvGJynVRCPnITkRGSE8VcW и всё будет корректно работать. Проблема в том, что из-за cloudflare система не может из профиля получить куку usess=*, без которой невозможно скачать корректный torrent-файл. uid и pass вы сможете найти в куках своего браузера, а usess указан в вашем профиле на LostFilm.TV
- Исправлены мелкие ошибки
23.03.2014 — 0.9.5:
Изменение для lostfilm.tv. В связи с тем, что в последнее время HD релизы выходят то в 720 то в 1080, и судя по всем — это так и останется, было принято решение немного переделать логику работы с данным трекером. Теперь качество 720 или 1080 выбирается мониторингом автоматически, в зависимости от того — в каком качестве серия была выложена, т.е. теперь не надо при добавлении выбирать качество 720 или 1080, это делается автоматом. Как показало изучение последних релизов — одновременно в 720 и 1080 релизов нет, по этому каких то проблем возникнуть не должно.
27.02.2014 — 0.9.4:
Добавлена поддержка трекера animelayer.ru
По многочисленным просьбам реализована возможность для каждой темы указывать свой путь для сохранения в torrent-клиенте при закачке (по-умолчанию качается в директорию, указанную в основных настройках).
09.02.2014 — 0.9.3:
На lostfilm.tv сменился адрес авторизации, по этому система могла последние дни ничего оттуда не качать. Так же на лосте появились сериалы в 1080p, но при этом пропали в 720p (возможно навсегда, как только удастся это подтвердить — настройка 720p из системы соответственно так же пропадёт), так что если вы качаете сериалы именно в этом качестве — вам необходимо их добавить заново!!!!
27.01.2014 — 0.9.2:
- Добавлен трекер casstudio.tv
- Добавлен трекер newstudio.tv
- Исправлены мелкие ошибки
20.01.2014 — 0.9.1:
- Добавлен трекер baibako.tv
- Исправлены мелкие ошибки
- Добавлено немного иконок, небольшие изменения в интерфейсе добавления торрентов
22.12.2013 — 0.9:
Всех с наступающими праздниками.
Наконец то я, по многочисленным просьбам, доделал поддержку торрент-клиентов. На данный момент это Transmission и Deluge, но только под linux/unix, т.к. работа строится на использовании консольного метода работы обоих клиентов.
Устранено несколько старых, мелких ошибок
16.11.2013 — 0.8.2:

js/user-func.js
$( document ).ready(function() 
{    // Скользящее меню
    $(".h-menu li").hover(
        function() {
            $(this).stop().animate({width: "235px"}, 500);
        },
        function() {
            if ($(this).hasClass("active")==false) {
                $(this).stop().animate({width: "27px"}, 500);
            }
        }
    );
    
    // Раскрывающийся список торрентов пользователя за которым ведётся мониторинг
    $(".user-torrent").click(function() {
        $(this).toggleClass("active");
        $(this).next().toggle();
        var id = $(this).attr('id');
        var date = new Date(new Date().getTime() + 60*1000);
        document.cookie="id="+id+"; path=/; expires="+date.toUTCString();
    });

    // Меню
    $(".h-menu li").click(function() {
        $(".h-menu li").stop().animate({width: "27px"}, 500);
        $(".h-menu li").removeClass("active");
        $(this).stop().animate({width: "235px"}, 500);
        $(this).addClass("active");
    });

    //Передаём пароль
    $("#enter").submit(function() {
        var $form = $(this),p = $form.find('input[name="password"]').val();
        ohSnap('Обрабатывается запрос...', 'yellow');
        $.post("action.php",{action: 'enter', password: p},
            function(data) {
                if (data.error)
                {
                    ohSnapX();
                    ohSnap(data.msg, 'red');
                }
                else {
                    ohSnapX();
                    ohSnap(data.msg, 'green');
                    showIndexContent();
                }
                console.log(data.error)
            }, "json"
        );
        
        return false;
    });

    //Передаём тему для мониторинга
    $("#torrent_add").submit(function()
    {
        var $form = $(this),
            s = $form.find('input[type=submit]'),
            n_f = $form.find('input[name="name"]'),
            n = $(n_f).val(),
            u_f = $form.find('input[name="url"]'),
            u = $(u_f).val(),
            p_f = $form.find('input[name="path"]'),
            p = $(p_f).val(); 
        
        if (u == '')
        {
            ohSnap('Вы не указали ссылку на тему!', 'red');
            return false;
        }
                                    
        ohSnap('Обрабатывается запрос...', 'yellow');
        $.post("action.php",{action: 'torrent_add', name: n, url: u, path: p},
            function(data) {
                ohSnapX();
                ohSnap(data, 'green');
                $(n_f).val('');
                $(u_f).val('');
            }
        );
        return false;
    });

    //Передаём сериал для мониторинга
    $("#serial_add").submit(function()
    {
        formError = "";
        var $form = $(this),
            s = $form.find('input[type=submit]'),
            t = $form.find('select[name="tracker"]').val(),
            n_f = $form.find('input[name="name"]'),
            n = $(n_f).val(),
            h_f = $form.find('input[name="hd"]'),
            p_f = $form.find('input[name="path"]'),
            p = $(p_f).val();

        h = $(h_f).val();
        qualitySelected = false;
        for (var i = 0; i < h_f.length; i++)
        {
            if (h_f[i].checked)
            {
                var $form = $(this), h = h_f[i].value;
                qualitySelected = true;
            }
        }

        if (t == '')
        {
            formError += "Вы не выбрали трекер!<br>";
        }
        
        if (n == '')
        {
            formError += "Вы не указали название сериала!<br>";
        }

        if (!qualitySelected)
        {
            formError += "Вы не выбрали качество!";
        }

        if (formError != "")
        {
            ohSnap(formError, 'red');
            return false;
        }

        ohSnap('Обрабатывается запрос...', 'yellow');
        $.post("action.php",{action: 'serial_add', tracker: t, name: n, hd: h, path: p},
            function(data) {
                ohSnapX();
                ohSnap(data, 'green');
                $(n_f).val('');
                $(h_f).removeAttr('checked');
            }
        );
        return false;
    });
    
    //Передаём данные для обновления
    $("#torrent_update").submit(function()
    {
        formError = "";
        var $form = $(this),
            s = $form.find('input[type=submit]'),
            i_f = $form.find('input[name="id"]'),
            id = $(i_f).val(),
            t_f = $form.find('input[name="tracker"]'),
            t = $(t_f).val(),
            n_f = $form.find('input[name="name"]'),
            n = $(n_f).val(),
            u_f = $form.find('input[name="url"]'),
            u = $(u_f).val(),
            update = $form.find('input[name="update"]').prop('checked'),
            p_f = $form.find('input[name="path"]'),
            p = $(p_f).val();
            h_f = $form.find('input[name="hd"]'),
            r_f = $form.find('input[name="reset"]').prop('checked');
            
        h = $(h_f).val();
        for (var i = 0; i < h_f.length; i++)
        {
            if (h_f[i].checked)
            {
                var $form = $(this), h = h_f[i].value
            }
        }
            
        if (u == '')
        {
            formError += "Вы не указали ссылку на тему!\n";
        }
        
        if (n == '')
        {
            formError += "Вы не указали название сериала!";
        }

        if (formError != "")
        {
            ohSnap(formError, 'red');
            return false;
        }

        ohSnap('Обрабатывается запрос...', 'yellow');
        $.post("action.php",{action: 'update', id: id, tracker: t, name: n, url: u, update: update, path: p, hd: h, reset: r_f},
            function(data) {
                ohSnapX();
                ohSnap(data, 'green');
            }
        );

        return false;
    });    

    //Передаём пользователя для мониторинга
    $("#user_add").submit(function()
    {
        formError = "";
        var $form = $(this),
            s = $form.find('input[type=submit]'),
            t = $form.find('select[name="tracker"]').val(),
            n_f = $form.find('input[name="name"]'),
            n = $(n_f).val();

        if (t == '')
        {
            formError += "Вы не выбрали трекер!\n";
        }
        
        if (n == '')
        {
            formError += "Вы не указали имя пользователя!";
        }

        if (formError != "")
        {
            ohSnap(formError, 'red');
            return false;
        }
        
        ohSnap('Обрабатывается запрос...', 'yellow');
        $.post("action.php",{action: 'user_add', tracker: t, name: n},
            function(data) {
                ohSnapX();
                ohSnap(data, 'green');
                $(n_f).val('');
            }
        );
        return false;
    });
    
    //Удаляем темы 
    $("#threme_clear").submit(function()
    {
        ohSnap('Обрабатывается запрос...', 'yellow');
        $.post("action.php",{action: 'threme_clear'},
            function(data) {
                ohSnapX();
                ohSnap(data, 'green');
                $.get("include/show_watching.php",
                    function(data) {
                        $('#content').empty().append(data);
                    }
                );
            }
        );
        return false;
    });

    //Передаём личные данные
    $("#credential").submit(function()
    {
        formError = "";
        var $form = $(this),
            b = $form.find('input[type=button]'),
            id = $form.find('input[name="id"]').val(),
            l = $form.find('input[name="log"]').val(),
            p = $form.find('input[name="pass"]').val(),
            passkey = $form.find('input[name="passkey"]').val();

        if (l == '')
        {
            formError += "Вы не указали логин!\n";
        }
        
        if (p == '')
        {
            formError += "Вы не указали пароль!";
        }

        if (formError != "")
        {
            ohSnap(formError, 'red');
            return false;
        }
                                    
        ohSnap('Обрабатывается запрос...', 'yellow');
        $.post("action.php",{action: 'update_credentials', id: id, log: l, pass: p, passkey: passkey},
            function(data) {
                ohSnapX();
                ohSnap(data, 'green');
                $(b).removeAttr('disabled');
            }
        );
        return false;
    });
        
    //Передаём настройки
    $("#setting").submit(function()
    {
        formError = "";
        var $form = $(this),
            s = $form.find('input[type=submit]'),
            serverAddress = $form.find('input[name="serverAddress"]').val();
            auth = $form.find('input[name="auth"]').prop('checked');
            proxy = $form.find('input[name="proxy"]').prop('checked');
            autoProxy = $form.find('input[name="autoProxy"]').prop('checked');
            proxyAddress = $form.find('input[name="proxyAddress"]').val();
            torrent = $form.find('input[name="torrent"]').prop('checked');
            torrentClient = $form.find('select[name="torrentClient"]').val();
            torrentAddress = $form.find('input[name="torrentAddress"]').val();
            torrentLogin = $form.find('input[name="torrentLogin"]').val();
            torrentPassword = $form.find('input[name="torrentPassword"]').val();
            pathToDownload = $form.find('input[name="pathToDownload"]').val();
            deleteDistribution = $form.find('input[name="deleteDistribution"]').prop('checked');
            deleteOldFiles = $form.find('input[name="deleteOldFiles"]').prop('checked');
            rss = $form.find('input[name="rss"]').prop('checked');
            debug = $form.find('input[name="debug"]').prop('checked');
        
        if (serverAddress == '')
        {
            formError += "Вы не указали адрес сервера TM.\n";
        }
        
        if (proxy == 'checked' && proxyAddress == '')
        {
            formError += "Вы не указали адрес proxy-сервера.\n";
        }
        
        if (torrent == 'checked' && torrentClient == ''  && torrentAddress == '' && pathToDownload == '')
        {
            formError += "Вы не указали настройки торрент-клиента.";
        }

        if (formError != "")
        {
            ohSnap(formError, 'red');
            return false;
        }

        ohSnap('Обрабатывается запрос...', 'yellow');
        $.post("action.php",{action: 'update_settings', serverAddress: serverAddress, 
            auth: auth, proxy: proxy, autoProxy: autoProxy, proxyAddress: proxyAddress, torrent: torrent,
            torrentClient: torrentClient, torrentAddress: torrentAddress, torrentLogin: torrentLogin,
            torrentPassword: torrentPassword, pathToDownload: pathToDownload,
            deleteDistribution: deleteDistribution, deleteOldFiles: deleteOldFiles, rss: rss, debug: debug},
            function(data) {
                ohSnapX();
                ohSnap(data, 'green');
                $(s).removeAttr('disabled');
            }
        );
        return false;
    });

    //Передаём пароль
    $("#change_pass").submit(function()
    {
        formError = "";
        var $form = $(this),
            s = $form.find('input[type=submit]'),
            p = $form.find('input[name="password"]').val(),
            p2 = $form.find('input[name="password2"]').val();
            
        if (p == '')
        {
            formError += "Пароль не может быть пустым.\n";
        }
        
        if (p != p2) 
        {
            formError += "Пароль и подтверждение должны совпадать.";
        }

        if (formError != "")
        {
            ohSnap(formError, 'red');
            return false;
        }
        
        ohSnap('Обрабатывается запрос...', 'yellow');
        $.post("action.php",{action: 'change_pass', pass: p},
            function(data) {
                if (data.error)
                {
                    ohSnapX();
                    ohSnap(data, 'green');
                }
                else
                    document.location.reload();
            }, "json"
        );
        return false;
    });
    
    //Вызов процедуры обновления
    $("#system_update").submit(function()
    {
        $('#system_update').empty().append('<div id="loader"></div>');
        
        $.post("action.php",{action: 'system_update'},
            function(data) {
                $('#system_update').empty().html(data);
            }
        );
    });

    //Сохраняем настройки уведомлений
    $("#notifier_settings").submit(function()
    {
        //debugger;        
        var table = document.getElementById('notifiers-table');

        var notifiersSettings = {}; 
        for (var i = 1, row; row = table.rows[i]; i++) {
            var notifier = {};
            notifier["group"] = parseInt(row.getAttribute('group'));

            var notifSelect = row.children[0].children[0];
            notifier["notifier"] = notifSelect.options[notifSelect.selectedIndex].value;

            notifier["address"] = row.children[1].children[0].value;
            notifier["sendUpdate"] = row.children[2].children[0].checked;
            notifier["sendWarning"] = row.children[3].children[0].checked;

            notifiersSettings[i - 1] = notifier;
        } 
        ohSnap('Обрабатывается запрос...', 'yellow');
        var settings = JSON.stringify(notifiersSettings);
        $.post("action.php",{action: 'updateNotifierSettings', settings: settings},
            function(data) {
                ohSnapX();
                ohSnap(data, 'green');
            }
        );           
        return false;
    });

});

//Подгрузка страниц
function show(name)
{
    if (name == 'check' || name == 'execution' || name == 'update')
        $('#content').empty().append('<div id="loader"></div>');

    $.get("include/"+name+".php",
        function(data) {
            $('#content').empty().append(data);
    });

	if (name == 'show_table')
	{
		window.clearTimeout(this.timeoutID);
		this.timeoutID = window.setTimeout(function(){ show('show_table') },7000);
	}
	else if (name == 'show_warnings')
	{
		window.clearTimeout(this.timeoutID);
		this.timeoutID = window.setTimeout(function(){ show('show_warnings') },7000);
	}
	else
	{
		window.clearTimeout(this.timeoutID);
		delete this.timeoutID;
	}
	
	return false;
}

//Развернуть/свернуть слой
function expand(id)
{
	var div = "#"+id;
	if ($(div).is(":hidden"))
		$(div).slideDown("slow");
	else 
		$(div).slideUp("slow");
	return false;
}

//Удаляем пользователя
function delete_user(id)
{
    if (confirm("Удалить?"))
    {
    	ohSnap('Обрабатывается запрос...', 'yellow');
    	$.post("action.php",{action: 'delete_user', user_id: id},
    		function(data) {
    			ohSnapX();
                        ohSnap(data, 'green');
    			$.get("include/show_watching.php",
            		function(data) {
            			$('#content').delay(3000).empty().append(data);
            		}
            	);
    		}
    	);
    	return false;
    }
}

//Удаляем тему из буфера
function delete_from_buffer(id)
{
    if (confirm("Удалить?"))
    {
    	ohSnap('Обрабатывается запрос...', 'yellow');
    	$.post("action.php",{action: 'delete_from_buffer', id: id},
    		function(data) {
    			ohSnapX();
                        ohSnap(data, 'green');
    			$.get("include/show_watching.php",
            		function(data) {
            			$('#content').delay(3000).empty().append(data);
            		}
            	);
    		}
    	);
    	return false;
    }
}

//Перемещаем тему из буфера в мониторинг постоянный
function transfer_from_buffer(id)
{
	ohSnap('Обрабатывается запрос...', 'yellow');
	$.post("action.php",{action: 'transfer_from_buffer', id: id},
		function(data) {
			ohSnapX();
                        ohSnap(data, 'green');
			$.get("include/show_watching.php",
        		function(data) {
        			$('#content').delay(3000).empty().append(data);
        		}
        	);
		}
	);
	return false;

}

//Передаём темы для скачивания
function threme_add(id, user_id)
{
        ohSnap('Обрабатывается запрос...', 'yellow');
	$.post("action.php",{action: 'threme_add', id: id, user_id: user_id},
		function(data) {
			if (data.error)
			{
    			ohSnapX();
                        ohSnap('Ошибка передачи данных<br/>Попробуйте ещё раз.', 'green');
			}
			else
			{
				$.get("include/show_watching.php",
					function(data) {
						$('#content').empty().append(data);
					}
				);
				return false;				
			}
		}, "json"
	);
	return false;
}

//Удаляем мониторинг
function del(id, name)
{
    if (confirm('Удалить '+name+'?'))
    {
    	ohSnap('Обрабатывается запрос...', 'yellow');
    	$.post("action.php",{action: 'del', id: id},
    		function(data) {
    			ohSnapX();
                ohSnap(data, 'green');
    			$.get("include/show_table.php",
            		function(data) {
            			$('#content').delay(3000).empty().append(data);
            		}
            	);
    		}
    	);
    	return false;
    }
}

//Выводим логин/пароль в зависимости от выбранного в списке
function changefunc() 
{
    var select = document.getElementById("selectfunc");
    var selectedText = select.options[select.selectedIndex].text;
    var a = ['anidub.com', 'animelayer.ru', 'baibako.tv', 'casstudio.tv', 'kinozal.tv', 'lostfilm.tv', 'newstudio.tv', 'nnm-club.me', 'novafilm.tv', 'pornolab.net', 'rustorka.com', 'rutracker.org', 'tracker.0day.kiev.ua'];
    for (var i = 0; i < a.length; i++)
    {
        var e = a[i];
        var d;
        if (selectedText == e)
            d = "block";
        else
            d = "none";
        document.getElementById(e + "_label").style.display = d;
    }
}

//Меняем radiobutton
function changeField()
{
	var tracker = document.getElementById("tracker").value;
    if (tracker == 'baibako.tv' || tracker == 'newstudio.tv' || tracker == 'novafilm.tv')
        $('#changedField').empty().append('<span class="quality"><input type="radio" name="hd" value="0"> SD<br /><input type="radio" name="hd" value="1"> HD 720<br /><input type="radio" name="hd" value="2"> HD 1080</span>');
	if (tracker == 'lostfilm.tv' || tracker == 'lostfilm-mirror')
		$('#changedField').empty().append('<span class="quality"><input type="radio" name="hd" value="0"> SD<br /><input type="radio" name="hd" value="1"> Автовыбор HD 720/1080<br /><input type="radio" name="hd" value="2"> HD 720 MP4');
}

//Форма редактирования записи
function showForm(id)
{
    $.get("include/form.php", {'id': id},
        function(data) {
            $('.blok').empty().append(data);
        }
    );
    $(".coverAll").show();
}

//Помечаем новость как прочитанную
function newsRead(id)
{
    $.post("action.php", {action: 'markNews', id: id},
        function(data) {
            $('.'+id).removeClass();
        }
    );
}

$(".add-notifier-title").click(function() {
    //debugger;
    var maxGroup = 0;
    var table = document.getElementById('notifiers-table');
    for (var i = 1, row; row = table.rows[i]; i++) {
        var rowGroup = parseInt(row.getAttribute('group'));
        if (rowGroup > maxGroup)
            maxGroup = rowGroup;
    }
    maxGroup += 1;

    var availableNotifiersHtml = "";
    $.post("action.php", {action: 'getNotifierList'},
		function(data) {
            for (var notifier in data) {
                    availableNotifiersHtml += "<option value=\"" + data[notifier].Name + "\" >" + data[notifier].VerboseName + "</option>";
            }
            var table_row = '<tr class="notifierSettings" group="' + maxGroup + '">' +
                    '<td class="notifierSettings"><select id="sendService" name="sendService" style="width: 150px;">' + availableNotifiersHtml + '</select> </td>' +
                    '<td class="notifierSettings"><input type="text" name="sendAddress" value="" style="width: 300px;"> </td>' +
                    '<td class="notifierSettings"><input type="checkbox" name="sendUpdate" > </td>' +
                    '<td class="notifierSettings"><input type="checkbox" name="sendWarning" > </td>' +
                    '<td class="notifierSettings"><a class="delete" onclick="removeNotifierSetting(this)"></a> </td>' +
                    '</tr>';
            $(table_row).hide().insertAfter($('#notifiers-table tr:last')).removeClass("hide").addClass("show-row").hide().show('fast');
		}, "json"
	);
});

function removeNotifierSetting(btn) 
{
    var tr = btn.parentNode.parentNode;
    var group = tr.getAttribute('group');
    var notifSelect = tr.children[0].children[0];
    var notifier = notifSelect.options[notifSelect.selectedIndex].value;

    $.post("action.php", {action: 'removeNotifierSettings', notifierClass: notifier, group: group});

    tr.parentNode.removeChild(tr);
}


//Обновляем основную страницу без перезагрузки содержимого
function showIndexContent()
{
    $.post("action.php", {action: 'getIndexPage'},
        function(data) {
            $('#index_content').empty().append(data.content);
            
            if (data.type == 'auth') {
                FocusOnInput('password');
            }
            else if (data.type == 'main') {
                show('show_table');
            }
        }, "json"
    );
}

//Переводим фокус на переданное поле ввода 
function FocusOnInput(fieldName) {
    document.getElementById(fieldName).focus();
}